kongswap_adaptor/src/canister.rs

```diff
@@ -142,6 +142,12 @@ impl<A: AbstractAgent> TreasuryManager for KongSwapAdaptor<A> {
     async fn withdraw(&mut self, request: WithdrawRequest) -> TreasuryManagerResult {
         self.check_state_lock()?;
 
+        // We refresh the external custodian balances, as it could
+        // be unknown to the treasury manager, whether an external
+        // trader has swapped their tokens on the pool and consequently
+        // changes the balances or not.
+        self.refresh_balances().await;
```

Comment:
Note that this can interfere with another call to `refresh_balances` from the periodic task. 

I don't think this is a problem since in the worst case scenario we get the balances wrongly, and then we'd have to retry.